### PR TITLE
Adding Bitly and fixing another issue related to broken external links

### LINKED ISSUE #231: Integrate Bitly for lengthy links
image

For links that are lengthy and breaking the design, integrate bitly and shorten them

### PR DESCRIPTION
**Pull request by bansalnagesh**
Fixes #231

### CODE REVIEW

**Comment by bansalnagesh** [comment]
image

If its a large link, it will use bitly

image

**Comment by sradu** [comment]
Cross post:

I don't like that we have to pay for bit.ly. How about we try another approach and if the link is super big we should only the domain part? eg http://docs.google.com/... instead of that long google docs strirng.

**Comment by bansalnagesh** [comment]
image

File: app/views/projects/show.html.erb [hunk 1]
```diff
@@ -352,7 +352,10 @@
 
               <% if @project.links.present? %>
                 <%= project_panel_item(title: 'Helpful links') do %>
-                  <%= simple_format Rinku.auto_link(sanitize(@project.links), :all, 'class="text-indigo-500" rel="nofollow noopener noreferrer"'), { class: 'mb-3' }, sanitize: false %>
+                  <%#= simple_format (Rinku.auto_link(sanitize(@project.links), :all, 'class="text-indigo-500" rel="nofollow noopener noreferrer"')do |url| "#{URI.parse(url).host}\n\n\n\n" end), { class: 'mb-3' }, sanitize: false %>
+                  <div>
+                    <%= simple_format (Rinku.auto_link(sanitize(@project.links), :all, 'class="text-indigo-500" rel="nofollow noopener noreferrer"')do |url| "#{URI.parse(url).host}" end), { class: 'mb-1' }, sanitize: false %>
```
Comment (sradu) [inline app/views/projects/show.html.erb:357]: spacing after do

File: app/views/projects/show.html.erb [hunk 2]
```diff
@@ -352,7 +352,10 @@
 
               <% if @project.links.present? %>
                 <%= project_panel_item(title: 'Helpful links') do %>
-                  <%= simple_format Rinku.auto_link(sanitize(@project.links), :all, 'class="text-indigo-500" rel="nofollow noopener noreferrer"'), { class: 'mb-3' }, sanitize: false %>
+                  <%#= simple_format (Rinku.auto_link(sanitize(@project.links), :all, 'class="text-indigo-500" rel="nofollow noopener noreferrer"')do |url| "#{URI.parse(url).host}\n\n\n\n" end), { class: 'mb-3' }, sanitize: false %>
```
Comment (sradu) [inline app/views/projects/show.html.erb:355]: remove commented out lines.

File: Gemfile.lock [hunk 3]
```diff
@@ -101,6 +101,8 @@ GEM
       aws-eventstream (~> 1.0, >= 1.0.2)
     bcrypt (3.1.13)
     bindex (0.8.1)
+    bitly (2.0.0)
```
Comment (sradu) [inline Gemfile.lock:104]: remove bit.ly

**Pull request merged**